great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -841,7 +842,11 @@ def execution_engine_type(self) -> Type[SqlAlchemyExecutionEngine]:
     def get_engine(self) -> sqlalchemy.engine.Engine:
         if self.connection_string != self._cached_connection_string or not self._engine:
             try:
-                self._engine = sqlalchemy.create_engine(self.connection_string)
+                if isinstance(self.connection_string, ConfigStr):
+                    connection_string = self.connection_string.get_config_value()
```

Comment:
I'm wondering if we could always have a config string so call `get_config_value()` would always work and sometimes the connection string resolution is a no-op. I think you don't do this so we still get the pydantic validation on the plain connection string? Leaving it as is is ok with me but just wanted to bring it up to make sure i understand.

Reply:
Yes, exactly. Also, this will error if it can't find a config variable that matches the template, with the same error message a usage would get if this happened in the old style datasources.